fuel/datasets/binarized_mnist.py

```diff
@@ -1,19 +1,14 @@
 # -*- coding: utf-8 -*-
 import logging
 import os
-from collections import OrderedDict
-
-import numpy
 
 from fuel import config
-from fuel.datasets import IndexableDataset
-from fuel.utils import do_not_pickle_attributes
+from fuel.datasets import H5PYDataset
 
 logger = logging.getLogger(__name__)
 
 
-@do_not_pickle_attributes('indexables')
-class BinarizedMNIST(IndexableDataset):
+class BinarizedMNIST(H5PYDataset):
```

Comment:
I'm not sure how that would be implemented either, I'm curious to see his opinion on that.